sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeFixedWidthAggregationMap.java

```diff
@@ -243,9 +243,8 @@ public void printPerfMetrics() {
    * Note that this destroys the map, and as a result, the map cannot be used anymore after this.
    */
   public UnsafeKVExternalSorter destructAndCreateExternalSorter() throws IOException {
-    UnsafeKVExternalSorter sorter = new UnsafeKVExternalSorter(
+    return new UnsafeKVExternalSorter(
       groupingKeySchema, aggregationBufferSchema,
       SparkEnv.get().blockManager(), map.getPageSizeBytes(), map);
```

Comment:
ok. I see, we cannot directly insert records to the external sorter returned by `destructAndCreateExternalSorter`. We need to use merge.